File: pset6/credit/credit.py
import cs50
import math

while True:
    num = cs50.get_int("Number: ")
    if num >= 0:
        break

# number of digits
digits = int(math.log10(num)+1)


# sum digits
tot, first, second = 0, 0, 0

for i in range(1, digits+1):
    n = int(num % 10)

    # find the first digit
    if i == digits:
        first = n
    # find the second digit
    elif i == digits - 1:
        second = n

    if i % 2 == 0:
        n *= 2
        if n >= 10:
            r = int(n % 10)
            n = int(n / 10)
            tot += n + r
        else:
            tot += n
    else:
        tot += n
    num /= 10

if first == 3 and (second == 4 or second == 7) and digits == 15 and tot % 10 == 0:
    print("AMEX")
elif first == 5 and second < 6 and digits == 16 and tot % 10 == 0:
    print("MASTERCARD")
elif first == 4 and (digits == 13 or digits == 16) and tot % 10 == 0:
    print("VISA")
else:
    print("INVALID")

File: pset1/credit/credit.c
# include <stdio.h>
# include <cs50.h>
# include <math.h>

int main(void)
{
    // Get user input
    long long num;
    do
    {
        num = get_long_long("Number: ");
    }
    while (num < 0);

    // number of digits
    int digits = log10(num)+1;

    // sum digits
    int sum = 0;
    int n, first, second;
    for (int i=1; i <= digits; i++){
        n = num%10;

        // find first digit
        if (i == digits){
            first = n;
        }

        // find second digit
        else if (i == digits - 1){
            second = n;
        }

        if (i%2 ==0){
            n *= 2;
            if (n >= 10){
                int r = n % 10;
                n = n/10;
                sum += n+r;
            }
            else {
                sum += n;
            }
        }
        else {
            sum += n;
        }
        num /= 10;
    }

    if (first == 3 && (second == 4 || second == 7) && digits == 15 && sum%10 == 0){
        printf("AMEX\n");
    }
    else if (first == 5 && second < 6 && digits == 16 && sum%10 == 0){
        printf("MASTERCARD\n");
    }
    else if (first == 4 && (digits == 13 || digits == 16) && sum%10 == 0){
        printf("VISA\n");
    }
    else {
        printf("INVALID\n");
    }

}